src/utils/mlx_utils.c

```diff
@@ -14,6 +14,46 @@
 
 //= Privae =//
 
+#define GETLINE_BUFF 1280
+/**
+ * Function to read a file stream line by line.
+ * Since getline is not a c standard function it does not exist on windows.
+ * There for I created this function which is somewhat similarly to getline.
+ * 
+ * @param file File stream to read from.
+ * @return Allocated result string or NULL if the EOF is encountered and no characters
+ * have been read. If an error occurs, NULL is returned.
+ */
+char* mlx_getline(FILE *file)
+{
+	if (file == NULL)
+		return (NULL);
+
+	char* str;
+	size_t size = 1;
+	if (!(str = calloc(1, sizeof(char) * size)))
+		return (NULL);
```

Comment:
Is it possible to instead use realloc to initialize the allocation? Perhaps using a `do { ... } while(...)` could work.

Reply:
Realloc does allocate memory if the ptr is null:
```
If ptr is a null pointer, realloc() shall be equivalent to malloc() for the specified size.
```

Reply:
No longer have the first alloc as a side effect of the rewrite in d8bf221